python/tk_multi_workfiles/entity_models/extended_model.py

```diff
@@ -0,0 +1,280 @@
+        if entity_type not in self._entity_types:
+            return None
+        # If the model is empty, just bail out
+        if not self.rowCount():
+            return None
+        parent_list = [self.invisibleRootItem()]
+        while parent_list:
+            parent = parent_list.pop()
+            for row_i in range(parent.rowCount()):
+                item = parent.child(row_i)
+                if self.canFetchMore(item.index()):
+                    self.fetchMore(item.index())
+                entity = self.get_entity(item)
+                # If dealing with an item holding the entity type we're looking
+                # for, we don't add it to the list of items to explore. We return
+                # it if the id is the one we're looking for.
+                if not entity or entity["type"] != entity_type:
+                    if item.hasChildren():
+                        parent_list.append(item)
+                elif entity["id"] == entity_id:
+                    return item
+        return None
+
+    def item_from_field_value_path(self, field_value_list):
+        """
+        Retrieve an item from a list of field values identifying its path.
+
+        This allows to retrieve an item in an updated model from a list of
+        collected field values representing its path.
+
+        Full or partial matches are performed: if the item represented by the
+        given value list is not present in the model anymore, the last item
+        matched from the value list is returned.
+
+        :param field_value_list: A list of field values for the path from the
+                                 root to the item.
+        """
+        if not self.rowCount():
+            return None
+        parent = self.invisibleRootItem()
+        for field_value in field_value_list:
+            for row_i in range(parent.rowCount()):
+                item = parent.child(row_i)
+                if self.canFetchMore(item.index()):
+                    self.fetchMore(item.index())
+                value = item.data(self.SG_ASSOCIATED_FIELD_ROLE)
+                if value == field_value:
+                    parent = item
+                    break
+        return parent
+
+    def get_item_field_value_path(self, item):
+        """
+        Return a list of field values identifying the absolute path to the given item.
+
+        This can be collected and used later to retrieve the path to the item in an
+        updated model.
+
+        :returns: A list of field values for the path from the root to the item.
+        """
+        current_item = item
+        values = []
+        while current_item:
+            values.append(
+                current_item.data(self.SG_ASSOCIATED_FIELD_ROLE)
+            )
+            current_item = current_item.parent()
+        # Reverse the list we return.
+        return values[::-1]
+
+    def _get_key_for_field_data(self, field, sg_data):
```

Comment:
i think this should be moved down to the deferred model - it doesn't seem to be used anywhere in the extended model.

Generally speaking, it would be great with a more more info around this cache - perhaps in the deferred model class docstring? Explaining what is being cached, how it's being different from the caching already taking place in he sg model and why it's needed!

Reply:
moved.